src/collects/seashell/backend/project.rkt

```diff
@@ -316,13 +323,69 @@
           (hash-remove! locked-projects name) #t]
         [else (raise (exn:project (format "Could not unlock ~a!" name) (current-continuation-marks)))]))))
 
+;; (get-headers main-file)
+;; Produces a list of the local (i.e. the user's) header files included by a program, without .h
+;;
+;; Arguments:
+;;  main-file - The .c file being compiled
+;;  file-dir  - The directory containing main-file
+;;
+;; Returns:
+;;  A list of the .h files included by a program, with the .h extension stripped
+;; Raises:
+;;  exn:project if an included header is not a .h file
+;; TODO: need to clean up the subprocess and ports?
+(define/contract (get-headers main-file file-dir)
```

Comment:
You are going to want to clean up/close the I/O ports you get back from subprocess here.  The GC will not automatically close OS-level ports.